client/product-details/index.js

```diff
@@ -1,6 +1,35 @@
+/* global jQuery */
 /**
  * Internal dependencies
  */
-import bnplSiteMessaging from './bnpl-site-messaging';
+import { initializeBnplSiteMessaging } from './bnpl-site-messaging';
 
-bnplSiteMessaging();
+const bnplPaymentMessageElement = initializeBnplSiteMessaging();
+
+jQuery( function ( $ ) {
+	$( '.quantity input' ).on( 'change', function ( event ) {
+		const newQuantity = event.target.value;
+		const price = window.wcpayStripeSiteMessaging.price;
+		bnplPaymentMessageElement.update( {
+			amount: parseInt( price, 10 ) * newQuantity,
+		} );
+	} );
+
+	// Handle BNPL payment message changes for product variation.
+	$( '.single_variation_wrap' ).on( 'show_variation', function (
+		event,
+		variation
+	) {
+		const quantity = $( '.quantity input' ).val();
+		window.wcpayStripeSiteMessaging.price = variation.display_price * 100;
```

Comment:
This sort of price manipulation will not work for some of currencies.
Note how we supply the generic product price on PHP side,  we need something similar for variations as well (a hook perhaps?).

Reply:
Just to make sure, are you referring to adjustment for zero decimal currencies via [this function](https://github.com/Automattic/woocommerce-payments/blob/13c124ab8b5a92f5ae4a48bc8f6fe98b9608723a/includes/class-wc-payments-utils.php#L119) ?

Reply:
Yep

Reply:
Thanks for the detailed catch here, @kalessil 🙌 

Here is my attempt to take care of this 
80a578d

Currently none of the supported BNPL currencies are of zero decimal, however you can test this by momentarily changing `$currency_code` to `jpy` or `krw` (small case ) [on this line](https://github.com/Automattic/woocommerce-payments/blob/80a578d969921491455e24468e0bab0259626f40/includes/class-wc-payments-payment-method-messaging-element.php#L52-L53) , and then check the multiplier that appears in browser tools on `window.wcpayStripeSiteMessaging` . 

<img width="700" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/4162931/7dbe5b3f-9e36-48cc-bae8-3c6d4d040975">

With currency back to USD .. it looks like this 

<img width="700" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/4162931/37b951cd-9375-4416-a163-34de64fe1574">

Reply:
Nope, still a fragile solution. See my hints [here](https://github.com/Automattic/woocommerce-payments/commit/80a578d969921491455e24468e0bab0259626f40#r120707661).

Reply:
Thanks for pointing to the pitfall and additional hints! 
Hopefully this commit fixes it - 9489c82 

I have done the following: 
1. Create a `variationsPriceList`  by using `$product->get_children` and then getting prepared prices for each of them using the existing `prepare_amount()` function. 
2. Pass this  within the JS script data, and on the JS side, do a lookup of `variationId` to get the normalized cost. 

Looks like this for USD: 

<img width="500" src="https://github.com/Automattic/woocommerce-payments/assets/4162931/3d5fb6b5-7013-40c7-95eb-b9292c094354">


Like this for JPY : 

<img width="334" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/4162931/2bdc02c2-c06c-4913-a8a9-50ae61665ded">

Let me know if this is okay @kalessil